File: Map/Z_InterpretorWCB/src/statement/PrintStatement.java
package statement;

import domain.IDictionary;
import domain.IHeap;
import domain.IList;
import domain.PrgState;
import expression.IExpression;

public class PrintStatement implements IStatement
{
	private IExpression expr;
	
	public PrintStatement(IExpression e)
	{
		expr = e;
	}

	@Override
	public PrgState execute(PrgState ps) 
	{
		IList<Integer> list = ps.getOut();
		IDictionary<String,Integer> dict = ps.getSymbolTable();
		IHeap<Integer,Integer> heap = ps.getHeap();
		int result = expr.eval(dict,heap);
		list.add(result);
		return null;
	}
	
	@Override
	public String toString()
	{
		return "Print("+expr+")";
	}

}

File: Map/Map/Map/statement/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;
using Map.expression;

namespace Map.statement
{
    class PrintStatement : IStatement
    {
        private IExpression expr;

        public PrintStatement(IExpression ex)
        {
            expr = ex;
        }

        public ProgramState execute(ProgramState ps)
        {
            ps.outList.add(expr.eval(ps.symbolTable));
            return null;
        }

        public override String ToString()
        {
            return "Print(" + expr + ")";
        }
    }
}
